servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp

```diff
@@ -1201,9 +1201,12 @@ void RenderForwardClustered::_update_volumetric_fog(Ref<RenderSceneBuffersRD> p_
 
 	if (p_render_buffers->has_custom_data(RB_SCOPE_FOG)) {
 		Ref<RendererRD::Fog::VolumetricFog> fog = p_render_buffers->get_custom_data(RB_SCOPE_FOG);
-		//validate
-		if (p_environment.is_null() || !environment_get_volumetric_fog_enabled(p_environment) || fog->width != target_width || fog->height != target_height || fog->depth != get_volumetric_fog_depth()) {
+
+		if (p_environment.is_null() || !environment_get_volumetric_fog_enabled(p_environment)) {
 			p_render_buffers->set_custom_data(RB_SCOPE_FOG, Ref<RenderBufferCustomDataRD>());
+
+		} else if (p_environment.is_valid() || environment_get_volumetric_fog_enabled(p_environment) || fog->width != target_width || fog->height != target_height || fog->depth != get_volumetric_fog_depth()) {
+			p_render_buffers->set_custom_data(RB_SCOPE_FOG, previous_fog);
```

Comment:
It looks like this never resizes the fog buffer. You would end up just stretching the old fog buffer to the new viewport size.

This line of code is just setting the fog buffer to itself, and then it results in the condition below that creates the new fog buffer never being called.

Reply:
To properly fix this, we need to allow the buffer to resize without flickering. Stopping the buffer from resizing at run time is not the proper solution